### ISSUE TITLE
When creating a new item, scroll to its position

### ISSUE THREAD

**Issue by manolo** [vaadin/vaadin-crud#60]

**Comment by manolo** [comment]
There are several issues to handle this

- `scrollToIndex` is marked as private, hence we need to use `_scrollToIndex` which is protected
- Crud component only can compute the index of the item to pass as parameter to `_scrollToIndex` in the case of items, but not for dataProvider.
- Filters and sorters make imposible to scroll to the item given it's index, hence we need to reset those before scrolling
- Reset filters and sorters requires transvers the DOM to find helpers and set value and direction properties. When we have custom text-fields in filter helpers, it makes more difficult to handle it.
- When user has a custom grid, we can not rely that he is using helpers for sort/filter or a custom approach

Therefore, there is no a crud out-of-the-box solution covering all usecases. It means that user needs to handle it in their app.

I have created a [branch](https://github.com/vaadin/vaadin-crud/compare/fix/60?expand=1) demonstrating how to reset filters and sorters and move the scroll. It is using protected API and needs a change in crud-grid implementation, so probably it's not a good idea to merge it wdy @jouni @alvarezguille

**Comment by manolo** [comment]
Probably it could be a good idea that crud offers some methods for dealing with grid quirks and expose something like `scrollToIndex` `resetSorters` `resetFilters` , that should improve DX. Out of 1.0 scope I think.

**Comment by jouni** [comment]
Yes, out of scope for now, as it seems complicated. Needs more discussion.
